internal/input/report.go

```diff
@@ -0,0 +1,207 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.`
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package input
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	ecc "github.com/enterprise-contract/enterprise-contract-controller/api/v1alpha1"
+	"github.com/hashicorp/go-multierror"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/enterprise-contract/ec-cli/internal/evaluator"
+	"github.com/enterprise-contract/ec-cli/internal/format"
+	"github.com/enterprise-contract/ec-cli/internal/policy"
+	"github.com/enterprise-contract/ec-cli/internal/version"
+)
+
+type Input struct {
+	FilePath     string
+	Violations   []evaluator.Result
+	Warnings     []evaluator.Result
+	Successes    []evaluator.Result
+	Success      bool
+	SuccessCount int
+}
+
+type Report struct {
+	Success       bool `json:"success"`
+	created       time.Time
+	FilePaths     []Input                          `json:"filepaths"`
+	Policy        ecc.EnterpriseContractPolicySpec `json:"policy"`
+	EcVersion     string                           `json:"ec-version"`
+	Data          any                              `json:"-"`
+	EffectiveTime time.Time                        `json:"effective-time"`
+	PolicyInput   [][]byte                         `json:"-"`
+}
+
+type summary struct {
+	FilePaths []inputSummary `json:"filepaths"`
+	Success   bool           `json:"success"`
+	Key       string         `json:"key"`
+}
+
+type inputSummary struct {
+	FilePath        string              `json:"name"`
+	Success         bool                `json:"success"`
+	Violations      map[string][]string `json:"violations"`
+	Warnings        map[string][]string `json:"warnings"`
+	Successes       map[string][]string `json:"successes"`
+	TotalViolations int                 `json:"total_violations"`
+	TotalWarnings   int                 `json:"total_warnings"`
+	TotalSuccesses  int                 `json:"total_successes"`
+}
+
+// TestReport represents the standardized TEST_OUTPUT format.
+// The `Namespace` attribute is required for the appstudio results API. However,
+// it is always an empty string from the ec-cli as a way to indicate all
+// namespaces were used.
+type TestReport struct {
+	Timestamp string `json:"timestamp"`
+	Namespace string `json:"namespace"`
+	Successes int    `json:"successes"`
+	Failures  int    `json:"failures"`
+	Warnings  int    `json:"warnings"`
+	Result    string `json:"result"`
+	Note      string `json:"note,omitempty"`
+}
+
+// Possible formats the report can be written as.
+const (
+	SummaryMarkdown = "summary-markdown"
```

Comment:
Drop this and `junit` it they're not supported?

Reply:
Ack. Removed.